lib/scoped/update.js

```diff
@@ -2,10 +2,45 @@ module.exports = update
 
 var find = require('./find')
 
+var merge = require('lodash/merge')
+var toId = require('pouchdb-hoodie-api/utils/to-id')
+
 function update (type, api, objectsOrIds, change) {
   return find(type, api, objectsOrIds)
 
   .then(function (storedObjects) {
-    return api.update(objectsOrIds, change)
+    if (!storedObjects) {
+      return Promise.resolve(storedObjects)
+    }
+
+    // objectsOrIds may contain updated information for the respective storedObject
+    // (e.g. _deleted = true), so incorporate all information in the objectOrId into
+    // the storedObject
+    var updatedObjects = updateObjects(objectsOrIds, storedObjects)
+
+    return api.update(updatedObjects, change)
   })
 }
+
+function updateObjects (objectsOrIds, storedObjects) {
+  if (!Array.isArray(objectsOrIds)) {
+    return updateObject(objectsOrIds, storedObjects)
+  }
+
+  var idIndex = -1
+  return storedObjects.map(function (storedObject) {
+    do {
+      var id = toId(objectsOrIds[++idIndex])
+    } while (id !== storedObject.id)
```

Comment:
I must admit I don’t quite get your implementation (it was also a long long day :/ ). Could you explain it to me?

My thinking is that we walk through `storedObjects`, ignore the errors, return the strings, and if it’s an `object` apply the properties from `objectsOrIds[index]`, something like this:


```js
storedObjects.map(function (storedObject, index) {
  // pass through errors from find
   if (storedObject instanceof Error) {
    return objectOrId
  }
  
  // return storedObject if objectOrId is an id
  if (typeof objectOrIds[index] === 'string') {
    return storedObject
  }
  
  // otherwise apply changes from passed object at current index
  return updateObject(storedObject, objectOrIds[index]) 
})
```

Does that make sense? Please let me know if I miss something, I likely do :)